### DISCUSSION TITLE
Change all occurences of `uid` with `id`

### DISCUSSION THREAD

**Discussion by bidoubiwa** [meilisearch/product#389] [Feedback & Feature Proposal]
Since `uid` is the same in definition as `id` I was wondering if for simplicity we could not just change the names of the field to `id`. 

I also, and this is only a feeling based on no evidence, that `uid` is used in a context of a simplified generated [`uuid`](https://en.wikipedia.org/wiki/Universally_unique_identifier). So I would think that `uid` would represent a sort of generated unique slug like `123abc45`.

**Comment by dichotommy** [comment]
I agree that 1) `uid` is unnecessarily complicated in some cases 2) `uid` suggests a unique value generated by Meilisearch. I'm not sure that changing all `uid` to `id` is the best solution though.

As far as I am aware, **indexes** and **tasks** are the two objects in Meilisearch that have the `uid` property. Am I forgetting anything else? 🤔 

What stands out to me is that index `uid` is user-set, whereas task `uid` is automatically generated by Meilisearch. I think it could benefit us to distinguish between these. For example, **if we replaced index `uid` with `name`, I think that would immediately suggest a property set by the user**.

(I believe there is already a `name` property attached to the index object but AFAIK it does nothing)

On the other hand, I think task `uid` could probably remain as is. Changing it to `id` would indeed be more simple. But from a documentation perspective, it could create confusion with the primary key, which is sometimes referred to as the document identifier or document id (and for inference, automatically looks for a field called `id` in documents).

WDYT @bidoubiwa ?

**Reply by dichotommy** [reply]
Edit: I forgot about dumps which also have `uid`. 😅

**Reply by bidoubiwa** [reply]
I like `name` for index! It does indeed imply that it is user set very well. As in SQL a name must be unique (even if in this case you can update it but that is not very important I think). 

In the other cases, tasks and dumps, `uid` are indeed generated! Now the `uid` of the dumps is more in the sense of a `uuid` -> `  "uid": "20200929-114144097"` where the `uid` of a task is more of an incremental `id` => `"uid": 1`, `"uid": 2`

**Reply by brunoocasali** [reply]
I like `name` for the index as well, our MongoDB friends use it too: https://www.mongodb.com/docs/manual/reference/method/db.createCollection/#mongodb-method-db.createCollection

But my feeling when I read `name` in a json object, I feel that I (as a user), can add anything here like:

```json
{
    "name": "my beautiful index",
    "createdAt": "2022-04-14T19:59:29.226971255Z",
    "updatedAt": "2022-04-14T19:59:29.228628255Z",
    "primaryKey": "id"
}
```

```json
{
    "uid": "my-beautiful-index",
    "createdAt": "2022-04-14T19:59:29.226971255Z",
    "updatedAt": "2022-04-14T19:59:29.228628255Z",
    "primaryKey": "id"
}
```

Without respecting any guidelines like these https://docs.meilisearch.com/learn/core_concepts/indexes.html#index-uid.

So, for me, the only change I would consider will be the `tasks`, summarizing:

- indexes - `uid`
- dumps - `uid`
- tasks - `id`

**Reply by bidoubiwa** [reply]
Yes I did not considered this `"name": "my beautiful index",`.
`id`, `uid` and `name` have their flaws. `uid` implies auto-generated complex number, `id` implies incremental and generated number  and `name` let you think that you can name it with spaces (or other weirdness).

I looked at how mango tackle this issue, they created a [`naming restriction`](https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limit-Restriction-on-Collection-Names) page that I found on your link. It explains how a name should be formatted.
